framework/encode/capture_manager.cpp

```diff
@@ -113,90 +112,129 @@ CaptureManager::~CaptureManager()
-
         // Initialize logging to report only errors (to stderr).
         util::Log::Settings stderr_only_log_settings;
         stderr_only_log_settings.min_severity            = util::Log::kErrorSeverity;
         stderr_only_log_settings.output_errors_to_stderr = true;
         util::Log::Init(stderr_only_log_settings);
 
+        // WIP WIP -- FIRST WINS for settings
         // Get capture settings which can be different per capture manager.
-        CaptureSettings settings(GetInstanceFunc()->GetDefaultTraceSettings());
+        default_settings_ = api_capture_singleton->GetDefaultTraceSettings();
+        capture_settings_ = api_capture_singleton->GetDefaultTraceSettings();
 
         // Load log settings.
-        CaptureSettings::LoadLogSettings(&settings);
+        CaptureSettings::LoadLogSettings(&capture_settings_);
 
         // Reinitialize logging with values retrieved from settings.
         util::Log::Release();
-        util::Log::Init(settings.GetLogSettings());
+        util::Log::Init(capture_settings_.GetLogSettings());
 
         // Load all settings with final logging settings active.
-        CaptureSettings::LoadSettings(&settings);
+        CaptureSettings::LoadSettings(&capture_settings_);
 
         GFXRECON_LOG_INFO("Initializing GFXReconstruct capture layer");
         GFXRECON_LOG_INFO("  GFXReconstruct Version %s", GFXRECON_PROJECT_VERSION_STRING);
 
-        CaptureSettings::TraceSettings trace_settings = settings.GetTraceSettings();
+        CaptureSettings::TraceSettings trace_settings = capture_settings_.GetTraceSettings();
         std::string                    base_filename  = trace_settings.capture_file;
 
         // Initialize capture manager with default settings.
-        success = GetInstanceFunc()->Initialize(base_filename, trace_settings);
+        success = Initialize(base_filename, trace_settings);
         if (!success)
         {
             GFXRECON_LOG_FATAL("Failed to initialize CaptureManager");
         }
     }
+
+    // Add the api capture managers to the list
+    GFXRECON_ASSERT(api_capture_singleton != nullptr);
+    auto manager_it = api_capture_managers_.find(api_capture_singleton);
+    if (manager_it == api_capture_managers_.end())
+    {
+        // Add the Api Capture Manager to this list, and point it back to this common manager.
+        auto inserted =
+            api_capture_managers_.insert(std::make_pair(api_capture_singleton, ApiInstanceRecord{ 1, destroyer }));
+        api_capture_singleton->SetCommonManager(this);
+        GFXRECON_ASSERT(inserted.second);
+        manager_it = inserted.first;
+
+        // WIP NOTE moved here from CaptureTracker::Initialize... DRY'r than putting it into the API specific
+        // CreateInstances
+        //          but I'm not satisfied it's the best place for it, or that we aren't being to clever lazily adding
+        //          it.
+        if ((capture_mode_ & kModeTrack) == kModeTrack)
```

Comment:
Why not just add this as part of whatever sets the common manager.  Like instead of `SetCommonManager` above, call `InitTrackingState` and it takes the common manger and does this state tracker creation as well.

Reply:
Didn't want to introduce a new virtual entry point, and this has to be revisited when we address the multiple dissimilar API's with different settings issue.